File: src/wsTripXML/Code/NDCClasses/NdcAuthenticate.vb
Imports System.Web.Services
Imports TripXMLMain
Imports System.Xml
Imports System.Xml.Serialization
Imports System.Threading
Imports System.Data
Imports CompressionExtension
Imports Microsoft.VisualBasic
Imports System.Text

Namespace wsTravelTalk


    Public Class ndcAuthenticate
        Inherits System.Web.Services.Protocols.SoapHeader

        Public POS As POS
        Public compressed As Boolean = True
    End Class

    Public Class POS
        Public Source As Source

        '<remarks/>
        Public TPA_Extensions As TPA_Extensions
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Source

        '<remarks/>
        Public RequestorID As RequestorID

        '<remarks/>
        Public Profile As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public PseudoCityCode As String

    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class RequestorID

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public ID As String

    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class TPA_Extensions

        '<remarks/>
        Public Provider As Provider

    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Provider

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("Name")>
        Public Name() As Name

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("System")>
        Public GDSSystem As String

        '<remarks/>
        Public Userid As String

        '<remarks/>
        Public Password As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Name

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public PseudoCityCode As String

        '<remarks/>
        <System.Xml.Serialization.XmlTextAttribute()>
        Public Value As String
    End Class

    '<remarks/>
    <System.Xml.Serialization.XmlRootAttribute(IsNullable:=False)>
    Public Class Ticket

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public RPH As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public StatusIndicator As String

        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public Exchange As Boolean = True

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public OmitInfant As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public OmitInfantSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlElementAttribute("PassangerInfo")>
        Public PassangerInfo() As PassangerDetails

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public TicketType As TicketTypeType

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public Number As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute(),
         System.ComponentModel.DefaultValueAttribute(False)>
        Public BoardingPass As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public FlightRefNumberRPHList As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public InfantOnly As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public InfantOnlySpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public IssueInvoice As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public IssueInvoiceSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public IssueItinerary As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public IssueItinerarySpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public ExchangeSpecified As Boolean

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public RemoteLocation As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public SpecialInstruction As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public IssueJointInvoice As Boolean = False

        '<remarks/>
        <System.Xml.Serialization.XmlIgnoreAttribute()>
        Public IssueJointInvoiceSpecified As Boolean
    End Class


    '<remarks/>
    Public Enum TicketTypeType

        '<remarks/>
        Electronic

        '<remarks/>
        Paper

        '<remarks/>
        MCO

        '<remarks/>
        Void
    End Enum


    '''<remarks/>
    Partial Public Class PassangerDetails

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public RPH As String

        '<remarks/>
        <System.Xml.Serialization.XmlAttributeAttribute()>
        Public Code As String

    End Class


End Namespace

File: src/wsTripXML/Code/NDCClasses/NdcAuthenticate.cs
using System.Xml;
using System.Xml.Serialization;

namespace wsTripXML.wsTravelTalk
{
    public class ndcAuthenticate : System.Web.Services.Protocols.SoapHeader
    {

        public POS POS;
        public bool compressed = true;
    }

    public class POS : Code.IPOS
    {
//        public Source Source;
        public TPA_Extensions TPA_Extensions;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Source
    {

        // <remarks/>
        public RequestorID RequestorID;

        // <remarks/>
        public string Profile;

        // <remarks/>
        [XmlAttribute()]
        public string PseudoCityCode;

    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class RequestorID
    {

        // <remarks/>
        [XmlAttribute()]
        public string ID;

        // <remarks/>
        [XmlAttribute(DataType = "anyURI")]
        public string URL;

        // <remarks/>
        [XmlAttribute()]
        public string Type;

        // <remarks/>
        [XmlAttribute()]
        public string Instance;

    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class TPA_Extensions : Code.ITPA_Extensions
    {

        // <remarks/>
        public Provider Provider;

    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Provider
    {

        // <remarks/>
        [XmlElement("Name")]
        public Name[] Name;

        // <remarks/>
        [XmlElement("System")]
        public string GDSSystem;

        // <remarks/>
        public string Userid;

        // <remarks/>
        public string Password;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Name
    {

        // <remarks/>
        [XmlAttribute()]
        public string PseudoCityCode;

        // <remarks/>
        [XmlText()]
        public string Value;
    }

    // <remarks/>
    [XmlRoot(IsNullable = false)]
    public class Ticket
    {

        // <remarks/>
        [XmlAttribute()]
        public string RPH;

        // <remarks/>
        [XmlAttribute()]
        public string StatusIndicator;

        [XmlAttribute()]
        public bool Exchange = true;

        // <remarks/>
        [XmlAttribute()]
        public bool OmitInfant = false;

        // <remarks/>
        [XmlIgnore()]
        public bool OmitInfantSpecified;

        // <remarks/>
        [XmlElement("PassangerInfo")]
        public PassangerDetails[] PassangerInfo;

        // <remarks/>
        [XmlAttribute()]
        public TicketTypeType TicketType;

        // <remarks/>
        [XmlAttribute()]
        public string Number;

        // <remarks/>
        [XmlAttribute()]
        [System.ComponentModel.DefaultValue(false)]
        public bool BoardingPass = false;

        // <remarks/>
        [XmlAttribute()]
        public string FlightRefNumberRPHList;

        // <remarks/>
        [XmlAttribute()]
        public bool InfantOnly = false;

        // <remarks/>
        [XmlIgnore()]
        public bool InfantOnlySpecified;

        // <remarks/>
        [XmlAttribute()]
        public bool IssueInvoice = false;

        // <remarks/>
        [XmlIgnore()]
        public bool IssueInvoiceSpecified;

        // <remarks/>
        [XmlAttribute()]
        public bool IssueItinerary = false;

        // <remarks/>
        [XmlIgnore()]
        public bool IssueItinerarySpecified;

        // <remarks/>
        [XmlIgnore()]
        public bool ExchangeSpecified;

        // <remarks/>
        [XmlAttribute()]
        public string RemoteLocation;

        // <remarks/>
        [XmlAttribute()]
        public string SpecialInstruction;

        // <remarks/>
        [XmlAttribute()]
        public bool IssueJointInvoice = false;

        // <remarks/>
        [XmlIgnore()]
        public bool IssueJointInvoiceSpecified;
    }


    // <remarks/>
    public enum TicketTypeType
    {

        // <remarks/>
        Electronic,

        // <remarks/>
        Paper,

        // <remarks/>
        MCO,

        // <remarks/>
        Void
    }


    /// <remarks/>
    public partial class PassangerDetails
    {

        // <remarks/>
        [XmlAttribute()]
        public string RPH;

        // <remarks/>
        [XmlAttribute()]
        public string Code;

    }


}